R/Private.R

```diff
@@ -115,3 +115,157 @@ nullToEmpty <- function(x) {
   x[is.null(x)] <- ""
   return(x)
 }
+
+
+
+makeDataExportable <- function(x,
+                               tableName,
+                               minCellCount = 5,
+                               databaseId = NULL) {
+  ParallelLogger::logTrace(paste0(" - Ensuring data is exportable: ", tableName))
+  if (is.null(x) || nrow(x) == 0) {
+    ParallelLogger::logTrace("  - Object has no data.")
+  }
+  
+  resultsDataModel <- getResultsDataModelSpecifications()
+  
+  if (!is.null(databaseId)) {
+    x <- x %>%
+      dplyr::mutate(databaseId = !!databaseId)
+  }
+  
+  fieldsInDataModel <- resultsDataModel %>%
+    dplyr::filter(.data$tableName == !!tableName) %>%
+    dplyr::pull(.data$fieldName) %>%
+    snakeCaseToCamelCase() %>%
+    unique()
+  
+  requiredFieldsInDataModel <- resultsDataModel %>%
+    dplyr::filter(.data$tableName == !!tableName) %>%
+    dplyr::filter(.data$isRequired == "Yes") %>%
+    dplyr::pull(.data$fieldName) %>%
+    snakeCaseToCamelCase() %>%
+    unique()
+  
+  primaryKeyInDataModel <- resultsDataModel %>%
+    dplyr::filter(.data$tableName == !!tableName) %>%
+    dplyr::filter(.data$primaryKey == "Yes") %>%
+    dplyr::pull(.data$fieldName) %>%
+    snakeCaseToCamelCase() %>%
+    unique()
+  
+  columnsToApplyMinCellValue <- resultsDataModel %>%
+    dplyr::filter(.data$tableName == !!tableName) %>%
+    dplyr::filter(.data$minCellCount == 'Yes') %>%
+    dplyr::pull(.data$fieldName) %>%
+    snakeCaseToCamelCase() %>%
+    unique()
+  
+  ParallelLogger::logTrace(paste0(
+    "  - Found in table ",
+    tableName,
+    " the following fields: ",
+    paste0(names(x), collapse = ", ")
+  ))
+  
+  presentInBoth <-
+    intersect(fieldsInDataModel, names(x))
+  presentInDataOnly <-
+    setdiff(names(x), fieldsInDataModel)
+  missingRequiredFields <-
+    setdiff(requiredFieldsInDataModel, presentInBoth)
+  
+  if (length(presentInDataOnly) > 0) {
+    warning(
```

Comment:
I would change this from a warning to a logInfo - our code should just be resilient to users adding optional fields but notifying them is useful